soccer/src/soccer/strategy/agent/position/goalie.cpp

```diff
@@ -16,70 +16,44 @@ rj_msgs::msg::RobotIntent Goalie::get_task() {
     }
 
     WorldState* world_state = this->world_state();  // thread-safe getter
-    bool needs_to_block = false;
-    auto block_pt = get_block_pt(world_state, needs_to_block);
-
-    if (needs_to_block) {
-        // create PathTargetMotionCommand, set goal position to block_pt
-        auto ptmc = rj_msgs::msg::PathTargetMotionCommand{};
-        ptmc.target.position = rj_convert::convert_to_ros(block_pt);
-
-        // make goalie face the ball
-        rj_geometry::Point ball_pos = world_state->ball.position;
-        auto face_pt = ball_pos;
-        ptmc.override_face_point = {rj_convert::convert_to_ros(face_pt)};
-
-        ptmc.ignore_ball = true;  // allow goalie to intersect ball's path
-
-        // update intent
-        intent.motion_command.path_target_command = {ptmc};
+    if (shot_on_goal_detected(world_state)) {
+        // TODO(Kevin): fix intercept planner's is_done, then add in logic to
+        // clear/pass ball once done intercepting
+        auto intercept_mc = rj_msgs::msg::InterceptMotionCommand{};
+        intercept_mc.target.x = 0.0;
+        intercept_mc.target.y = 0.1;
+        intent.motion_command.intercept_command = {intercept_mc};
+        intent.motion_command.name = "intercept";
```

Comment:
I like the .name addition. It makes the factory design pattern more viable also.

Reply:
In #1988 I actually moved the .name field to RobotIntent.msg instead of MotionCommand.msg, for easier conversion to C++ struct (see this [commit](https://github.com/RoboJackets/robocup-software/pull/1988/commits/ccd0b504c706a0991092940699db829862c4dbb2)).

I can move it back and be less lazy with the ROS->C++ conversion methods. But while working on the PR I realized the current planner architecture is a bit less easy to convert to use a Factory method than I first thought, so I decided to go with whatever was easiest for me to add for debugging purposes.

Where do you think it should go?